pkg/ocp/machinesets.go

```diff
@@ -87,6 +100,16 @@ func (msd *k8sMachineSetDeployer) GetWorkerNodeImage(workerNodeList []string, ma
 			return "", errors.Wrapf(err, "error retrieving machine set %q", nodeName)
 		}
 
+		labels, _, _ := unstructured.NestedSlice(existing.Object, "spec", "template", "metadata", "labels")
+		for _, l := range labels {
+			label := l.(map[string]interface{})
+			value, _, _ := unstructured.NestedString(label, "machine.openshift.io/cluster-api-machine-role")
+
+			if strings.Compare(strings.ToLower(value), "worker") != 0 {
+				continue
+			}
+		}
```

Comment:
Isn't the `labels` field a map and not a slice? Shouldn't this work:

```suggestion
		labels, found, _ := unstructured.NestedStringMap(existing.Object, "spec", "template", "metadata", "labels")
		if found {
		    role := labels["machine.openshift.io/cluster-api-machine-role"]
		    if strings.Compare(strings.ToLower(role), "worker") != 0 {
		        continue
	             }
		}
```

Reply:
Yes @tpantelis it will work.